Bring string handling more in line with real Go

dominikh opened the pull request.

dmitshur commented:
LGTM.

The overall change is good and closes #3 (you should mention that). Docs seem clear and accurate to me. I've also tested it in my code, and it compiled and ran correctly.

dominikh committed:
don't sanitize strings, always deal with binary data

In Go, strings are usually considered immutable byte slices, and their
encoding doesn't matter, at least not when transmitting them as data.

JavaScript has a different idea, which leads to strings being sanitized
so that they contain valid UTF-8, by replacing invalid byte sequences
with replacement runes.

Since we're writing this package primarily for Go programmers, we want
it to behave as close to "real" Go as possible. That's why the top level
Send function is changed to receive binary data and return []byte, and
why Request.Send is changed to never pass strings to JS, only []byte.

This makes it impossible for someone to actually make use of
aforementioned sanitization for outgoing strings, but there's no reason
XHR itself should be doing it, and it can be solved in pure Go with a
loop.

See the discussion at https://github.com/dominikh/go-js-xhr/issues/3 for
more information.

dominikh committed:
make top-level Send only accept []byte

dmitshur commented on xhr.go:
```diff
@@ -58,6 +58,15 @@ const (
+//   req := xhr.NewRequest("POST", "http://example.com")
+//   req.ResponseType = xhr.ArrayBuffer
+//   req.Send([]byte("data"))
+//   b := js.Global.Get("Uint8Array").New(req.Response).Interface().([]byte)
```
While this is a logical place for this information to be, I do think it would be more visible and helpful overall to simply move this section to the end of package overview docs.

dominikh committed:
bring "low level API" closer to JS behaviour

Do not convert string to []byte, and explain how to recreate the
behaviour of xhr.Send.
